base/logging.jl

```diff
@@ -483,23 +483,30 @@ function handle_message(logger::SimpleLogger, level, message, _module, group, id
         logger.message_limits[id] = remaining - 1
         remaining > 0 || return
     end
-    levelstr = string(level)
```

Comment:
Changing this could end up a bit misleading for custom log levels.

Reply:
@c42f would you be OK with renaming `Warn` to `Warning`? The reason for this change is that all the other levels are nouns, and I prefer we print `Warning` over `Warn`.

Reply:
Yes, it's `Warn` vs `Warning` which is the sticking point here.  The upside of the current naming is that `Warn` is consistent with `@warn`, and all the standard levels are four or five characters long.

So it seems impossible to have complete consistency without renaming to `@warning` in which case I'd rather just leave it as it currently is.  Of course, we can just special case the printing more or less as you've done, but taking into account custom levels.  I agree it's not entirely satisfying.

Reply:
`Warning` seems very standarded to use, looking at random pictures of logging on google... Forcing the `levelstr` to be the same as the macro used seems a bit inflexible imo. Why can't we use `@warn` and print it as `Warning`?

Reply:
Certainly, we can print it differently for logging output, absolutely no problem with that. I thought the issue was about the named constant `Logging.Warn::LogLevel`.

For achieving "Warning" printed in the output we can have
* `string(Logging.Warn) == "Warning"` - seems too inconsistent to really consider.
* `Logging.Warn` -> `Logging.Warning` (which I thought was being suggested above). I'm not so keen on that, for the reasons of consistency with the spelling of `@warn`.
* Just special case the output formatting, as is done already.  This seems the best of the lot IMO, though not entirely satisfying.

Reply:
Perhaps I was confused about what @fredrikekre was suggesting. Sorry if I was misunderstanding.

Reply:
No probs, I really wish we could get together in the same room to discuss where this stuff is going.  It's super slow and easy to get confused / talk at cross purposes in text.

Option (2) is a reasonable way out if you want to try it @fredrikekre and see how it feels? I'm not really sure where the best compromise of (2) vs (3) lies. They both have some inconsistencies, unfortunately.